dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1704,17 +1705,87 @@ public Identifier getRelatedIdentifier(Contentlet contentlet,String relationship
         }
     }
 
+    private Optional<Boolean> checkAndRunDeleteAsWorkflow(final Contentlet contentletIn, final User user,
```

Comment:
This method and `checkAndRunDestroyAsWorkflow` have logic in common that can be extracted in a method an reused